File: internals/product/controller/dto/product.go
package dto

import "time"

type Product struct {
	ID          string    `json:"id"`
	Code        string    `json:"code"`
	Name        string    `json:"name"`
	ImageUrl    string    `json:"image_url"`
	Description string    `json:"description"`
	Price       float64   `json:"price"`
	Active      bool      `json:"active"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/interface/product.ts
export interface IProduct {
  id: string
  code: string
  name: string
  image_url: string
  description: string
  price: number
  active: boolean
  created_at: string
  updated_at: string
}

export interface IListProductRequest {
  search: string
  page: number
  size: number
  order_by: string
  order_desc: boolean
  take_all: boolean
}

export interface ICreateProductRequest {
  name: string
  description: string
  image: any
  price: number
}

export interface IUpdateProductRequest {
  name: string
  description: string
  image: any
  price: number
}
